tests/manifest.cc

```diff
@@ -57,19 +61,371 @@ test_yamlToJSON0()
 }
 
 
+/* -------------------------------------------------------------------------- */
+
+/** @brief Test that a simple descriptor can be parsed from JSON. */
+  bool
+test_parseManifestDescriptor0()
+{
+
+  flox::resolver::ManifestDescriptorRaw raw = R"( {
+    "name": "foo"
+  , "version": "4.2.0"
+  , "optional": true
+  , "packageGroup": "blue"
+  , "packageRepository": "nixpkgs"
+  } )"_json;
+
+  flox::resolver::ManifestDescriptor descriptor( raw );
+
+  EXPECT( descriptor.name.has_value() );
+  EXPECT_EQ( * descriptor.name, "foo" );
+
+  /* Ensure this string was detected as an _exact_ version match. */
+  EXPECT( ! descriptor.semver.has_value() );
+  EXPECT( descriptor.version.has_value() );
+  EXPECT_EQ( * descriptor.version, "4.2.0" );
+
+  EXPECT( descriptor.group.has_value() );
+  EXPECT_EQ( * descriptor.group, "blue" );
+  EXPECT_EQ( descriptor.optional, true );
+
+  /* We expect this to be recognized as an _indirect flake reference_. */
+  EXPECT( descriptor.input.has_value() );
+  EXPECT( std::holds_alternative<nix::FlakeRef>( * descriptor.input ) );
+  auto flakeRef = std::get<nix::FlakeRef>( * descriptor.input );
+
+  EXPECT_EQ( flakeRef.input.getType(), "indirect" );
+
+  auto alias = flakeRef.input.attrs.at( "id" );
+  EXPECT( std::holds_alternative<std::string>( alias ) );
+  EXPECT_EQ( std::get<std::string>( alias ), "nixpkgs" );
```

Comment:
Seems like a few of these tests would be cleaner if you defined `operator==` on `ManifestDescriptor`

Reply:
Nonblocking